dir.targets

```diff
@@ -28,6 +28,15 @@
     <TargetFrameworkProfile></TargetFrameworkProfile>
     <!-- We set this property to avoid MSBuild errors regarding not setting TargetFrameworkProfile (see above line) -->
     <PortableNuGetMode>true</PortableNuGetMode>
+
+    <!--Start with NonUnixTests if we aren't on Windows and overwrite with a specific OS, if we know what it is-->
+    <TestFilterCategory Condition="'$(OS)' != 'Windows_NT'">-notrait category=nonunixtests</TestFilterCategory>
+    <TestFilterCategory Condition="'$(OS)' == 'OSX'">-notrait category=nonosxtests</TestFilterCategory>
+    <TestFilterCategory Condition="'$(OS)' == 'Windows_NT'">-notrait category=nonwindowstests</TestFilterCategory>
+    <TestFilterCategory Condition="'$(OS)' == 'Linux'">-notrait category=nonlinuxtests</TestFilterCategory>
+    <TestFilterCategory Condition="'$(OS)' == 'FreeBSD'">-notrait category=nonfreebsdtests</TestFilterCategory>
+    <TestFilterCategory Condition="'$(OS)' == 'NetBSD'">-notrait category=nonnetbsdtests</TestFilterCategory>
+    <TestCommandLine>$(TestCommandLine) $(TestFilterCategory)</TestCommandLine>
```

Comment:
Yes I'm cool with staging this in corefx and then prompting to buildtools.

As for the specific change I'm not entirely sure $(OS) is the right way to go here. Msbuild will currently only set it to Unix, OSX, or Windows (https://github.com/radical/msbuild/blob/fix-isosx/src/Shared/NativeMethodsShared.cs#L534). I think we probably should start passing it through the builds file as TestOSGroup or something like that which we can default to the OSGroup passed in at the root or we can start explicitly passing it.